dagfactory/dagbuilder.py

```diff
@@ -826,23 +762,33 @@ def build(self) -> Dict[str, Union[str, DAG]]:
         tasks_tuples = self.topological_sort_tasks(tasks)
         for task_name, task_conf in tasks_tuples:
             task_conf["task_id"]: str = task_name
-            operator: str = task_conf["operator"]
             task_conf["dag"]: DAG = dag
-            # add task to task_group
+            params: Dict[str, Any] = {k: v for k, v in task_conf.items() if k not in SYSTEM_PARAMS}
+
             if task_groups_dict and task_conf.get("task_group_name"):
                 task_conf["task_group"] = task_groups_dict[task_conf.get("task_group_name")]
-            # Dynamic task mapping available only in Airflow >= 2.3.0
-            if (task_conf.get("expand") or task_conf.get("partial")) and version.parse(AIRFLOW_VERSION) < version.parse(
-                "2.3.0"
-            ):
-                raise DagFactoryConfigException("Dynamic task mapping available only in Airflow >= 2.3.0")
 
-            # replace 'task_id.output' or 'XComArg(task_id)' with XComArg(task_instance) object
-            if task_conf.get("expand") and version.parse(AIRFLOW_VERSION) >= version.parse("2.3.0"):
-                task_conf = self.replace_expand_values(task_conf, tasks_dict)
-            params: Dict[str, Any] = {k: v for k, v in task_conf.items() if k not in SYSTEM_PARAMS}
-            task: Union[BaseOperator, MappedOperator] = DagBuilder.make_task(operator=operator, task_params=params)
-            tasks_dict[task.task_id]: BaseOperator = task
+            if "operator" in task_conf:
+                operator: str = task_conf["operator"]
+                # Dynamic task mapping available only in Airflow >= 2.3.0
+                if (task_conf.get("expand") or task_conf.get("partial")) and version.parse(
+                    AIRFLOW_VERSION
+                ) < version.parse("2.3.0"):
+                    raise DagFactoryConfigException("Dynamic task mapping available only in Airflow >= 2.3.0")
+
+                # replace 'task_id.output' or 'XComArg(task_id)' with XComArg(task_instance) object
+                if task_conf.get("expand") and version.parse(AIRFLOW_VERSION) >= version.parse("2.3.0"):
```

Comment:
```suggestion
                if task_conf.get("expand"):
```
wdyt, since we are checking the version in a 1 block above?

Reply:
That was a create cleanup suggestion, @pankajastro , I just improved this in 96b601c2b628d446d548efadd94060ea7deeb861